lib/core/src/Cardano/Wallet/Primitive/Model.hs

```diff
@@ -200,9 +201,18 @@ applyBlock !b (Wallet !u !pending _ s bh) =
         txs' = Map.fromList $ map
             (\(tx, meta) -> (txId @t tx, (tx, meta)))
             txs
+
+        -- In particular this invariant forbids Epoch Boundary Blocks
+        -- to be applied since they share slotId with a normal block.
+        --
+        -- This is in turn done to make sure our blockHeight calculation is
+        -- correct.
+        hdr = invariant "applyBlock: slotId of the block must be greater than\
+            \the slotId of the previously applied block." (b ^. #header)
+            (\h -> (h ^. #slotId) > (prevHdr ^.  #slotId))
```

Comment:
I would propose the following formatting:

```hs
        hdr = invariant
            ("applyBlock: "
                <> "slotId of the block must be greater than the slotId of the "
                <> "previously-applied block.")
            (b ^. #header)    
            (\h -> (h ^. #slotId) > (prevHdr ^.  #slotId))    
```

This way, it's clearer that the `invariant` function has three separate parameters.

Alternatively:

```hs
```hs
        hdr = invariant
            "applyBlock: \
                \slotId of the block must be greater than the slotId of the \
                \previously-applied block."
            (b ^. #header)    
            (\h -> (h ^. #slotId) > (prevHdr ^.  #slotId))    
```